From 5179f002bb19047f8f35667b0d28b51cfcbc5947 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:22:57 -0400
Subject: [x86] perf/intel: Fix build warning in intel_pmu_drain_pebs_nhm()

Message-id: <1408462094-14194-9-git-send-email-jolsa@redhat.com>
Patchwork-id: 87995
O-Subject: [PATCH RHEL7.1 BZ1131394 008/325] perf/x86/intel: Fix build warning in intel_pmu_drain_pebs_nhm()
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 92519bbc8af612975410def52bd462ca9af85cdb
Author: Peter Zijlstra <peterz@infradead.org>
Date: Fri Sep 20 13:08:50 2013 +0200

description
===========
Fengguang Wu reported this build warning:

    arch/x86/kernel/cpu/perf_event_intel_ds.c: In function 'intel_pmu_drain_pebs_nhm':
    arch/x86/kernel/cpu/perf_event_intel_ds.c:964:2: warning: format '%ld' expects argument of type 'long int', but argument 4 has type 'int'

Because pointer arithmetics result type is bitness dependent there's no natural
type to use here, cast it to long.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kernel/cpu/perf_event_intel_ds.c b/arch/x86/kernel/cpu/perf_event_intel_ds.c
index e79be19..d915996 100644
--- a/arch/x86/kernel/cpu/perf_event_intel_ds.c
+++ b/arch/x86/kernel/cpu/perf_event_intel_ds.c
@@ -957,7 +957,7 @@ static void intel_pmu_drain_pebs_nhm(struct pt_regs *iregs)
   */
  WARN_ONCE(top - at > x86_pmu.max_pebs_events * x86_pmu.pebs_record_size,
     "Unexpected number of pebs records %ld\n",
-    (top - at) / x86_pmu.pebs_record_size);
+    (long)(top - at) / x86_pmu.pebs_record_size);
 
  for (; at < top; at += x86_pmu.pebs_record_size) {
   struct pebs_record_nhm *p = at;
-- 
1.7.1